### PR TITLE
Read playlist with track media that are remote audio files

### PR DESCRIPTION
**Pull request by hugurp**
In this PR:

- [x] add event playerReady so we can get the instance of the player and pass it tracks (index.html)
- [x] add a new `filePlayer` component that plays from files (should support same actions as `youtubePlayer`)
- [x] be sure it works with remote files `https://site.com/path-to-file.mp3`
- [x] define clearely what type of media content is supported (ogg, mp3, aac, wav, aif, flac...) - maybe write in docs https://developer.mozilla.org/en-US/docs/Web/HTML/Supported_media_formats
- [x] serialize the data (a json with a channel info and track list) when imported in `PlayerData.vue`.
- [x] connect the components `filePlayer` with `ProviderPlayer` (track changed, play/pause actions, events - copy youtube ways if it makes sense)

Right now data models are based on r4:
- `track` model from r4 https://radio4000.firebaseio.com/tracks/-Kp-4o6aSBcYrV8iev2P.json
- `channel` model from r4 https://radio4000.firebaseio.com/channels/-JYEosmvT82Ju0vcSHVP.json

So to serialize a track we need a new key `mediaUrl` maybe `mediaUri`, to point to where the media is.
Then the key channel is unecessary, also url and youtube id.
```
id: -Kp-4o6aSBcYrV8iev2P
body | "@radio-gael, open hats, compressing bounce #am #pm"
channel | "-JYEosmvT82Ju0vcSHVP"
created | 1500018852619
title | "Junktion - Prologue"
url | "https://youtu.be/bcKXP0f2Hw4"
ytid | "bcKXP0f2Hw4"
```

# Features for next PR
- [x] ~~be sure it works with local files `~/music/path-to-file.ogg` (we cannot gain access on local folders, expect if we import the files, like in the dj )~~

### CODE REVIEW

**Comment by hugurp** [comment]
Right now I can't figure out where to send this `playerReady` event.

On the `connectedCallback` of the custom-element, and on the `mounted` hok of the playerData, it is send too early.

Result is that:
```
// index.html
player.addEventListener('playerReady', () => {
  var playerInstance = player.__vue_custom_element__.$children[0];
  // __vue_custom_element__ is undefined because playerReady event
  // is sent too early
```

**Comment by oskarrough** [comment]
I searched the vue-custom-element repo and found it emits an `vce-ready` event. In our custom element I listen to this event and send our own `playerReady`.

https://github.com/karol-f/vue-custom-element/blob/fc830934d86b599404c187bd9180183abc68fcc7/src/utils/createVueInstance.js#L100-L113

**Comment by hellerve** [comment]
This is amazing! What do we need to do still to make this work?

**Comment by oskarrough** [comment]
Proposing to the to-do list above:

- [ ] hide `__vue_custom_element` API from users. e.g. use `player.updatePlaylist()` directly
- [ ] maybe `fileUrl` instead of `mediaUrl`? Makes it obvious that it is only files as compared to youtube media
- [ ] document format of `playlist` object in readme

`setPlaylist` vs `updatePlaylist`? Technically it always overwrites.

File: src/FilePlayer.vue [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+		name: 'file-player',
+		props: {
+			autoplay: Boolean,
+			isPlaying: Boolean,
+			mediaUrl: String,
+			volume: Number
+		},
+		data() {
+			return {
+				continousPlay: true
```
Comment (oskarrough) [inline src/FilePlayer.vue]: typo, should be "continuousPlay"

File: src/FilePlayer.vue [hunk 2]
```diff
@@ -0,0 +1,96 @@
+<template>
+	<audio class="FilePlayer"
+		controls
+		:src="mediaUrl"
+		:autoplay="false">
```
Comment (oskarrough) [inline src/FilePlayer.vue]: should be `:autoplay="autoplay` so it binds to the prop

**Comment by oskarrough** [comment]
- [ ] make sure `yarn build` works

**Comment by hellerve** [comment]
Oh yeah!

**Pull request merged**